plugins/logging/pkg/gateway/drivers/management/cluster_driver.go

```diff
@@ -0,0 +1,21 @@
+package management
+
+import (
+	"context"
+
+	"github.com/rancher/opni/pkg/opensearch/opensearch"
+	"github.com/rancher/opni/plugins/logging/pkg/apis/loggingadmin"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ClusterDriver interface {
+	Name() string
+	AdminPassword(context.Context) ([]byte, error)
+	NewOpensearchClientForCluster(context.Context) *opensearch.Client
+	GetCluster(context.Context) (*loggingadmin.OpensearchClusterV2, error)
+	DeleteCluster(context.Context) error
+	CreateOrUpdateCluster(context.Context, *loggingadmin.OpensearchClusterV2, string, *corev1.LocalObjectReference) error
```

Comment:
The LocalObjectReference here should probably be an equivalent non-k8s api type (plus, should give these parameters names)
```suggestion
	CreateOrUpdateCluster(context.Context, *loggingadmin.OpensearchClusterV2, string, string) error
```